app/src/main/java/com/android/streetworkapp/model/event/EventViewModel.kt

```diff
@@ -1,11 +1,19 @@
 package com.android.streetworkapp.model.event
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
 
 open class EventViewModel(private val repository: EventRepository) : ViewModel() {
 
+  private val _uiState: MutableStateFlow<EventOverviewUiState> =
+      MutableStateFlow(EventOverviewUiState.Empty)
+
+  val uiState: StateFlow<EventOverviewUiState> = _uiState
+
```

Comment:
Question: I am unsure about the use of EventOverviewUiState, is there an advantage to use it instead of a List<Event>? The name of the attribute (uiState) could also become confusing, do you think we could perhaps change it to a more specific name?

Reply:
Could you maybe clarify what you find confusing about this? 
I use a sealed class first to define the states of our overview screen (empty vs not empty) which i think makes the code more readable and also to make it that i don't necessarily have to fetch the EventList from the viewModel which makes the code more clean as it reduces the amount of public variables or potential calls to the `getEvents` method

Reply:
This was clarified in person, thank you for your explanation! (I therefor will mark this as resolved)